packages/appsync-modelgen-plugin/src/visitors/appsync-dart-visitor.ts

```diff
@@ -999,6 +1146,18 @@ export class AppSyncModelDartVisitor<
     });
   }
 
+  protected getModelIdentifierFields(model: CodeGenModel): CodeGenField[] {
```

Comment:
nit: This method can work until the v1 transformer is deprecated(as `@key` is v1). In long term, this method can take advantage of the change in base visitor, where the model fields can be traversed to find pk field and corresponding pk type info (including pk type and sort key fields). The pk name can also be found as well.
```TypeScript
const primaryKeyField = model.fields.find(field => field.primaryKeyInfo)!;
const { sortKeyFields } = primaryKeyField.primaryKeyInfo!
```